File: minibank_web/src/main/webapp/resource/js/transfer/transferHistory.js
$.urlParam = function(name){
    var results = new RegExp('[\?&]' + name + '=([^&#]*)').exec(window.location.href);
    if (results==null){
       return null;
    }
    else{
       return results[1] || 0;
    }
};
$(function() {
	var contextPath = $('#contextPath').val();
	var transferApiUrl = $('#transferApiUrl').val();
	var searchCstmId = $.urlParam('cstmId');
	$("#cstmId").focus();
	$('#transferMenu').addClass('selected-menu');
	$("#btnSearch").on("click", function(){
		var cstmId = $('#cstmId').val();
		if(cstmId == null || cstmId == ""){
			alert("고객ID를 입력해주세요.");
			$('#cstmId').focus();
		}
		else{
			$('#transferHistoryList td').remove();
			retrieveTransferHistoryList(cstmId);
		}
	});
	if(searchCstmId != null && searchCstmId != ''){
		$('#cstmId').val(searchCstmId);
		$('#btnSearch').click();
	}
	function retrieveTransferHistoryList(cstmId){
		$.ajax({
			type : 'GET',
			url : transferApiUrl+'/transfer-history/rest/v0.8/'+ cstmId,
			contentType: 'application/json',
			datatype : 'json',
			xhrFields : {
				withCredentials : true
			},
			crossDmain: true,
			beforeSend : function(){
		        $('.wrap-loading').removeClass('display-none');
			},
			complete:function(){
		        $('.wrap-loading').addClass('display-none');
			},
			success : function(data) {
				for(var i=0; i < data.length; i++){
					appendTransferHistoryHtml(data[i].trnfDtm, data[i].wthdAcntNo
							,data[i].dpstAcntNo, data[i].rcvCstmNm, data[i].trnfAmt
							,data[i].rcvMm, data[i].sndMm, data[i].divCd, data[i].stsCd);
				}
				
			},
		    error: function (jqXHR, textStatus, errorThrown) {
		    	if(jqXHR.status == '417' || jqXHR.status == '500'){
			    	var responseText = jqXHR.responseText;
			    	var body = JSON.parse(responseText);
			    	alert(body.message);
		    	}
		    	else
		    		alert("[연결 오류]\n서버와 연결에 실패했습니다.");
				$(location).attr("href", contextPath + "/transfer/view/retrieveTransferHistory");
		    }
		});
	}
	function appendTransferHistoryHtml(trnfDtm, wthdAcntNo, dpstAcntNo, rcvCstmNm, trnfAmt, rcvMm, sndMm, divCd, stsCd){
		
		var html = "<tr style='text-align : center;'>"
			     + "<td>"+((divCd=="1")?"당행":"타행")+"</td>"
			     + "<td>"+trnfDtm+"</td>"
			     + "<td>"+wthdAcntNo+"</td>"
			     + "<td>"+dpstAcntNo+"</td>"
			     + "<td>"+rcvCstmNm+"</td>"
			     + "<td>"+trnfAmt+"</td>"
			     + "<td>"+rcvMm+"</td>"
			     + "<td>"+sndMm+"</td>"
			     + "<td>"+((stsCd=="1")?"이체 요청":((stsCd=="2")?"이체 실패":"이체 완료"))+"</td>"
				 + "</tr>";
		$('#transferHistoryList').append(html);
				
	}
});

File: minibank_transfer/src/main/java/com/minibank/transfer/domain/entity/TransferHistory.java
package com.minibank.transfer.domain.entity;

import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Data
public class TransferHistory {
    private String cstmId;
    private int seq;
    private String divCd;
    private String stsCd;
    private String dpstAcntNo;
    private String wthdAcntNo;
    private int wthdAcntSeq;
    private String sndMm;
    private String rcvMm;
    private String rcvCstmNm;
    private Long trnfAmt;
    private String trnfDtm;
    
    @Builder
	public TransferHistory(String cstmId, int seq, String divCd, String stsCd, String dpstAcntNo, String wthdAcntNo,
			int wthdAcntSeq, String sndMm, String rcvMm, String rcvCstmNm, Long trnfAmt, String trnfDtm) {
		super();
		this.cstmId = cstmId;
		this.seq = seq;
		this.divCd = divCd;
		this.stsCd = stsCd;
		this.dpstAcntNo = dpstAcntNo;
		this.wthdAcntNo = wthdAcntNo;
		this.wthdAcntSeq = wthdAcntSeq;
		this.sndMm = sndMm;
		this.rcvMm = rcvMm;
		this.rcvCstmNm = rcvCstmNm;
		this.trnfAmt = trnfAmt;
		this.trnfDtm = trnfDtm;
	}
}

